### ISSUE TITLE
SSH Keys tab stuck on 'Wait for device confirmation' if Mooltipass passed to VM

### ISSUE THREAD

**Issue by asashnov** [mooltipass/moolticute#388]
SSH Keys tab stuck on 'Wait for device confirmation' if Mooltipass passed to VM.

Moolticute 319d3d990633bae7935b3cc8ae411b421722cc9c

UPDATE 7 Apr 2019: rephrase steps to reproduce to be easier to follow (by meaning they remain the same)

####Prerequsites:
* Host machine: Ubuntu 16.04 x64
* VirtualBox: 5.1.38-dfsg-0ubuntu1.16.04.1 with Oracle Extension Pack 5.1.38r122592
* VirtualBox virtual machine: Windows 7 SP1 x64 (build 7601)

#### Steps to reproduce:

1. Run Linux host machine
2. Run VirtualBox virtual machine.
2. Connect Mooltipass Mini to USB. Mooltipass device should not be passed into VM now.
3. Enter PIN to unlock the device
4. Run Moolticute application on Linux host, press Ctrl+Shift+F1 to show "SSH Keys",  
   click "Load Keys".  
   "Send Data For: moolticute ssh keys [v]" prompt will appear on the device.  
   Do NOT confirm this prompt. 
5. Instead, within 15 seconds (in less than 15 seconds, before the prompt will be auto-canceled)  
  switch to VirtualBox VM window and pass the Mooltipass device inside of this virtual machine  
  (right-click on the USB icon on Right-bottom of VM window, and pass 'SE Mooltipass [0100]' device inside the virtual machine.

#### Actual result:

Moolticute GUI 'SSH Keys" tab will stuck forever on 'Wait for device confirmation' while all the other tabs 'Device Settings', 'Credentials', 'Files'... will switch and look normally.

When 'Enter Credentials Management Mode' is clicked on "Credentials" tab "Waiting For Device Confirmation" appears also in this tab. But it is still possible to switch to any other tab. When switched back to "Credentials", 'Waiting For Device Confirmation' promtp disappears from "Credentials" tab (but still shown in "SSH Keys" tab).

#### Additional information:

In 'Daemon Logs' It seems that moolticute daemon stopped responding at all on GUI requests:

```
DEBUG: ../moolticute/src/WSServerCon.cpp:56 - JSON API recv: {
    "data": {
        "service": "moolticute ssh keys"
    },
    "msg": "data_node_exists"
}

INFO: ../moolticute/src/AsyncJobs.cpp:98 - "Check if data service exists: moolticute ssh keys reqid: "
INFO: ../moolticute/src/MPDevice.cpp:6356 - service_exists success
DEBUG: ../moolticute/src/WSClient.cpp:138 - New message:  {"data":{"exists":true,"service":"moolticute ssh keys"},"msg":"data_node_exists"}
INFO: ../moolticute/src/SSHManagement.cpp:125 - Running "/home/alex/mp/build-Moolticute-Desktop_Qt_5_9_7_GCC_64bit-Debug/mc-agent" ("--output_progress", "cli", "-c", "list")
DEBUG: ../moolticute/src/WSServerCon.cpp:1065 - Send card db metadata
DEBUG: ../moolticute/src/WSServerCon.cpp:1089 - Sended card db metadata
DEBUG: ../moolticute/src/WSServer.cpp:74 - New connection
DEBUG: ../moolticute/src/WSServerCon.cpp:56 - JSON API recv: {"msg":"get_data_node","data":{"service":"Moolticute SSH Keys"},"client_id":"75f68fd2-a676-48f0-b3ff-060d8e2c82e5"}
INFO: ../moolticute/src/AsyncJobs.cpp:98 - "Ask for data node for service: Moolticute SSH Keys fallback_service:  reqid: "
DEBUG: ../moolticute/src/MPDevice.cpp:128 - > Retry command:  "READ_32B_IN_DN (0xc1)"
DEBUG: ../moolticute/src/MPDevice.cpp:234 - "PLEASE_RETRY (0xc4)"  was received for a packet that was sent due to a timeout, not resending
WARNING: ../moolticute/src/MPDevice_linux.cpp:161 - Error receiving data:  Input/Output Error
DEBUG: ../moolticute/src/WSServerCon.cpp:56 - JSON API recv: {
    "msg": "get_random_numbers"
}

DEBUG: ../moolticute/src/WSServerCon.cpp:56 - JSON API recv: {
    "data": {
        "delay_after_key_enabled": true
    },
    "msg": "param_set"
}

DEBUG: ../moolticute/src/WSServerCon.cpp:56 - JSON API recv: {
    "data": {
        "want_data": false
    },
    "msg": "start_memorymgmt"
}

DEBUG: ../moolticute/src/WSServerCon.cpp:56 - JSON API recv: {
    "data": {
        "want_data": false
    },
    "msg": "start_memorymgmt"
}

DEBUG: ../moolticute/src/WSServerCon.cpp:56 - JSON API recv: {
    "data": {
        "want_data": false
    },
    "msg": "start_memorymgmt"
}
```

**Comment by raoulh** [comment]
What are you trying to do with the virtual machine here?? I do not follow you steps to reproduce this.

Also it is stuck because of an I/O error. It's written in the log:
`MPDevice_linux.cpp:161 - Error receiving data:  Input/Output Error`

**Comment by asashnov** [comment]
Probably libusb has a way to detect the device removal or something. In this case, moolticuted can wait until the device appears again.

**Comment by raoulh** [comment]
No it does not work like that

**Comment by limpkin** [comment]
current status? still happens?

**Comment by asashnov** [comment]
No, it doesn't get stuck forever now. Just for 1 minute or two, then Moolticute GUI falls to the right state "Connect Your Mooltipass Device".

Tested again on Linux Ubuntu 16.06
commit 449d76eb5f7789060a8dbbd0f3d5f28c63e6cc9b (Fri Apr 5 2019)

Daemon log changed compared to the previous issue, there is 'hidraw' instead libusb:

```
DEBUG: ../moolticute/src/WSServerCon.cpp:60 - JSON API recv: {
    "data": {
        "service": "moolticute ssh keys"
    },
    "msg": "data_node_exists"
}

INFO: ../moolticute/src/AsyncJobs.cpp:93 - "Check if data service exists: moolticute ssh keys reqid: "
INFO: ../moolticute/src/MPDevice.cpp:6568 - service_exists success
DEBUG: ../moolticute/src/WSClient.cpp:143 - New message:  {"data":{"exists":true,"service":"moolticute ssh keys"},"msg":"data_node_exists"}
INFO: ../moolticute/src/SSHManagement.cpp:125 - Running "/home/alex/mp/build-Moolticute-Desktop_Qt_5_9_7_GCC_64bit-Debug/mc-agent" ("--output_progress", "cli", "-c", "list")
DEBUG: ../moolticute/src/WSServerCon.cpp:542 - Send card db metadata
DEBUG: ../moolticute/src/WSServerCon.cpp:566 - Sended card db metadata
DEBUG: ../moolticute/src/WSServer.cpp:74 - New connection
DEBUG: ../moolticute/src/WSServerCon.cpp:60 - JSON API recv: {"msg":"get_data_node","data":{"service":"Moolticute SSH Keys"},"client_id":"4082f0df-2c91-453c-8138-79c950f913e9"}
INFO: ../moolticute/src/AsyncJobs.cpp:93 - "Ask for data node for service: Moolticute SSH Keys fallback_service:  reqid: "
DEBUG: ../moolticute/src/MPDevice.cpp:195 - > Retry command:  "READ_32B_IN_DN (0x00c1)"
DEBUG: ../moolticute/src/MPDevice.cpp:325 - "PLEASE_RETRY (0x00c4)"  was received for a packet that was sent due to a timeout, not resending
WARNING: ../moolticute/src/MPDevice_linux.cpp:96 - Failed to read from device:  Input/output error
DEBUG: ../moolticute/src/UsbMonitor_linux.cpp:58 - Node:  "/dev/hidraw3"
DEBUG: ../moolticute/src/UsbMonitor_linux.cpp:59 - Action:  "remove"
DEBUG: ../moolticute/src/MPManager.cpp:156 - Disconnecting:  "/dev/hidraw3"
DEBUG: ../moolticute/src/WSServer.cpp:143 - Mooltipass disconnected
DEBUG: ../moolticute/src/WSClient.cpp:143 - New message:  {"msg":"mp_disconnected"}
DEBUG: ../moolticute/src/DbBackupsTrackerController.cpp:257 - Db tracking is not available for fw < 1.2
DEBUG: ../moolticute/src/UsbMonitor_linux.cpp:58 - Node:  "/dev/hidraw4"
DEBUG: ../moolticute/src/UsbMonitor_linux.cpp:59 - Action:  "remove"
DEBUG: ../moolticute/src/MPManager.cpp:163 - "/dev/hidraw4"  is not connected.
```
